pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -209,8 +209,12 @@ rec {
 
           # This parameter is sometimes a string, sometimes null, and sometimes a list, yuck
           configureFlags = let inherit (lib) optional elem; in
-            (/**/ if lib.isString configureFlags then [configureFlags]
-             else if configureFlags == null      then []
+            (/**/ if lib.isString configureFlags then builtins.trace
+               "String `configureFlags` is deprecated since 18.09. Please use a list of flags, each a string."
+               [configureFlags]
+             else if configureFlags == null then builtins.trace
```

Comment:
I'm not sure `null` should be disallowed, since it has a special meaning because of the `__ignoreNulls` feature in Nix. For example, you can add an attribute like
```
configureFlags = if stdenv.isDarwin then ["blabla"] else null;
```
and this won't change the derivation on non-Darwin platforms (so doesn't cause a rebuild).

Reply:
@edolstra Currently because of the way configurePlatforms is handled, it's already a list anyways, so there is no risk of mass rebuilds.

In general, I rather default everything to `[]`, `false`, etc (or conversely filter out everything that isn't needed) so that the no-rebuild trik happens automatically without needing to write awkward things at the package level.